File: src/main/java/io/github/kuroka3/playermanager/Event/ConnectEvent.java
package io.github.kuroka3.playermanager.Event;

import io.github.kuroka3.playermanager.Class.ManagedPlayer;
import io.github.kuroka3.playermanager.PlayerManager;
import io.github.kuroka3.playermanager.Utils.JSONFile;
import org.bukkit.ChatColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerLoginEvent;
import org.json.simple.JSONObject;

import java.time.*;

public class ConnectEvent implements Listener {

    @EventHandler
    public void onConnect(PlayerLoginEvent e) {

        try {

            JSONFile playerJson = new JSONFile(PlayerManager.getPlugin(PlayerManager.class).getDataFolder() + "/players.json");

            JSONObject jsonObject = playerJson.getJSONObject();

            JSONObject userObject = (JSONObject) jsonObject.get(e.getPlayer().getUniqueId().toString());

            if(userObject == null) {
                JSONObject object = new JSONObject();

                object.put("name", e.getPlayer().getName());
                object.put("warns", 0);
                object.put("ban", false);
                object.put("banre", "");
                object.put("banid", "");
                object.put("mute", false);
                object.put("temp", false);
                object.put("tounban", "-1");

                jsonObject.put(e.getPlayer().getUniqueId().toString(), object);
            } else if(!userObject.get("name").equals(e.getPlayer().getName())) {
                userObject.put("name", e.getPlayer().getName());

                jsonObject.put(e.getPlayer().getUniqueId().toString(), userObject);
            }

            playerJson.saveJSONFile(jsonObject);

        } catch (Exception exception) {
            exception.printStackTrace();
        }

        try {
            ManagedPlayer p = new ManagedPlayer(e.getPlayer(), new JSONFile(PlayerManager.getPlugin(PlayerManager.class).getDataFolder() + "/players.json"));

            if (p.isBan()) {
                if(p.isTemp()) {
                    if(p.isTounabn(LocalDateTime.now())) {
                        p.unban();
                    } else {
                        LocalDateTime tounban = p.getTounban();

                        LocalDate tounbanDate = tounban.toLocalDate();
                        LocalTime tounbanTime = tounban.toLocalTime();

                        Period pr = Period.between(LocalDate.now(), tounbanDate);
                        Duration dr = Duration.between(LocalTime.now(), tounbanTime);

                        int[] nokori = {pr.getYears(), pr.getMonths(), pr.getDays(), dr.toHoursPart(), dr.toMinutesPart(), dr.toSecondsPart()};

                        for (int i = 0; i < nokori.length; i++) {
                            if(nokori[i]<0) nokori[i] = nokori[i]*-1;
                        }

                        e.disallow(PlayerLoginEvent.Result.KICK_BANNED,
                                ChatColor.RED + "당신은 이 서버에서 차단되었습니다\n\n" +
                                        ChatColor.RED + nokori[0] + "년 " + nokori[1] + "개월 " + nokori[2] + "일 " + nokori[3] + "시간 " + nokori[4] + "분 " + nokori[5] + "초 남음\n\n" +
                                        ChatColor.GRAY + "Reason: " + ChatColor.WHITE + p.getBanReason() +
                                        ChatColor.GRAY + "\n\nBan ID: " + ChatColor.WHITE + p.getBanid());
                    }
                } else {
                    e.disallow(PlayerLoginEvent.Result.KICK_BANNED, ChatColor.RED + "당신은 이 서버에서 차단되었습니다\n\n" + ChatColor.GRAY + "Reason: " + ChatColor.WHITE + p.getBanReason() + ChatColor.GRAY + "\n\nBan ID: " + ChatColor.WHITE + p.getBanid());
                }
            }
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }
}

File: src/main/kotlin/io/github/kuroka3/playermanager/Event/ConnectEvent.kt
package io.github.kuroka3.playermanager.Event

import io.github.kuroka3.playermanager.Class.ManagedPlayer
import io.github.kuroka3.playermanager.PlayerManager
import io.github.kuroka3.playermanager.Utils.Language
import io.github.kuroka3.playermanager.Utils.SettingsManager
import net.kyori.adventure.text.Component.text
import net.kyori.adventure.text.format.TextColor
import org.bukkit.OfflinePlayer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerLoginEvent
import org.json.simple.JSONObject
import java.time.*

class ConnectEvent : Listener {
    @EventHandler
    fun onConnect(e: PlayerLoginEvent) {
        try {
            if (e.result != PlayerLoginEvent.Result.ALLOWED) {
                return
            }

            val playerJson = PlayerManager.instance.playerJSONFile

            if(!playerJson.isFile || playerJson.isEmpty) {
                playerJson.createNewFile()
                playerJson.saveJSONFile(JSONObject())
            }

            val tempobj: JSONObject = playerJson.jsonObject!!

            val userObject: JSONObject? = tempobj[e.player.uniqueId.toString()] as JSONObject?

            if(userObject == null) {
                val `object` = JSONObject()

                `object`["warns"] = 0
                `object`["ban"] = false
                `object`["banre"] = ""
                `object`["banid"] = ""
                `object`["mute"] = false
                `object`["temp"] = false
                `object`["tounban"] = "-1"
                `object`["lang"] = SettingsManager["defaultLanguage", "en-us"]

                tempobj[e.player.uniqueId.toString()] = `object`
            } else if (userObject["name"] != e.player.name) {
                userObject["name"] = e.player.name
                tempobj[e.player.uniqueId.toString()] = userObject
            }

            playerJson.saveJSONFile(tempobj)

        } catch (ex: Exception) {
            ex.printStackTrace()
        }

        try {
            val p = ManagedPlayer(e.player as OfflinePlayer, PlayerManager.instance.playerJSONFile)

            if(p.ban) {
                if(p.temp) {
                    if(p.isToUnban(LocalDateTime.now())) {
                        p.unban(true)
                    } else {
                        e.disallow(PlayerLoginEvent.Result.KICK_BANNED,
                            text("${Language[p.lang, "player.banned"]}\n\n${p.getNokori(LocalDateTime.now(), Language[p.lang, "player.tempban"])}\n\n").color(
                                TextColor.color(0x00ff5555)
                            ).append(
                                text("Reason: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(p.banre).color(TextColor.color(0x00ffffff))).append(
                                text("\n\nBan ID: ").color(TextColor.color(0x00aaaaaa))).append(
                                text(p.banid).color(TextColor.color(0x00ffffff))))
                    }
                } else {
                    e.disallow(PlayerLoginEvent.Result.KICK_BANNED,
                        text("${Language[p.lang, "player.banned"]}\n\n")
                            .color(TextColor.color(0x00ff5555)).append(
                                text("Reason: ").color(TextColor.color(0x00aaaaaa))
                            ).append(
                                text(p.banre)
                                    .color(TextColor.color(0x00ffffff))
                            ).append(
                                text("\n\nBan ID: ").color(TextColor.color(0x00aaaaaa))
                            ).append(
                                text(p.banid)
                                    .color(TextColor.color(0x00ffffff))
                            ))
                }
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
            e.disallow(PlayerLoginEvent.Result.KICK_OTHER, text(Language[SettingsManager["defaultLanguage", "en-us"].toString(), "connect.failed"]).color(TextColor.color(0x00ff55555)))
        }
    }
}
